static/js/disaster_dashboard/ranking_section.tsx

```diff
@@ -0,0 +1,165 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for the ranking section of the disaster dashboard
+ */
+
+import _ from "lodash";
+import React from "react";
+
+import { DISASTER_EVENT_INTENSITIES, DISASTER_EVENT_TYPES } from "./constants";
+import { DisasterEventPoint, DisasterType } from "./types";
+
+const RANKING_ITEMS_COUNT = 5;
+
+interface RankingUnitInfo {
+  title: string;
+  prop: string;
+  ranking: DisasterEventPoint[];
+}
+
+interface RankingSectionPropType {
+  disasterEventPoints: DisasterEventPoint[];
+  selectedDisaster: DisasterType;
+  selectedIntensityProp: string;
+  onIntensityPropSelected: (prop: string) => void;
+}
+
+export function RankingSection(props: RankingSectionPropType): JSX.Element {
+  const rankedDisasterCounts = getRankedDisasterCounts(
+    props.disasterEventPoints
+  );
+  const rankingUnits = getRankingUnits(
+    props.disasterEventPoints,
+    props.selectedDisaster
+  );
+  const addSelectionCss =
+    !_.isEmpty(DISASTER_EVENT_INTENSITIES[props.selectedDisaster]) &&
+    DISASTER_EVENT_INTENSITIES[props.selectedDisaster].length > 1;
+  const getRankingUnitClassName = (prop: string) => {
+    if (!addSelectionCss) {
+      return "ranking-unit-no-selection";
+    }
+    return prop === props.selectedIntensityProp
+      ? "ranking-unit-selected"
+      : "ranking-unit";
+  };
+  return (
+    <div className="ranking-section">
+      {props.selectedDisaster === DisasterType.ALL && (
+        <>
+          <h3>Count of Events</h3>
+          {rankedDisasterCounts.map(({ disaster, count }) => {
+            return (
+              <div key={"count-" + disaster} className="ranking-unit-item">
+                {disaster}: {count}
+              </div>
+            );
+          })}
+        </>
+      )}
+      {(!_.isEmpty(rankingUnits) ||
+        props.selectedDisaster !== DisasterType.ALL) && (
+        <>
+          <h3>Rankings</h3>
+          {rankingUnits.map((rankingUnit) => {
+            return (
+              <div
+                onClick={() => props.onIntensityPropSelected(rankingUnit.prop)}
+                className={getRankingUnitClassName(rankingUnit.prop)}
+                key={rankingUnit.prop}
+              >
+                <div className={"ranking-unit-title"}>{rankingUnit.title}</div>
+                {rankingUnit.ranking
+                  .slice(0, RANKING_ITEMS_COUNT)
+                  .map((event) => {
+                    return (
+                      <div
+                        key={`${rankingUnit.prop}-${event.placeDcid}`}
+                        className="ranking-unit-item"
+                      >
+                        {event.placeName}: {event.intensity[rankingUnit.prop]}
+                      </div>
+                    );
+                  })}
+              </div>
+            );
+          })}
+        </>
+      )}
+    </div>
+  );
+}
+
+function getRankedDisasterCounts(
```

Comment:
Add comments for high level logic for these functions?

Reply:
If time allowed, would be nice to add simple test for these helper functions, so new people can update the code with more confidence.

Reply:
done